lib/byron/test/integration/Test/Integration/Byron/Scenario/API/Transactions.hs

```diff
@@ -440,6 +453,85 @@ scenario_TRANS_CREATE_07 = it title $ \ctx -> do
   where
     title = "TRANS_CREATE_07 - Deleted wallet"
 
+scenario_RESTORE_01
+    :: forall (n :: NetworkDiscriminant) t.
+        ( DecodeAddress n
+        , EncodeAddress n
+        , PaymentAddress n ByronKey
+        )
+    => (Context t -> IO ApiByronWallet)
+    -> SpecWith (Context t)
+scenario_RESTORE_01 fixtureSource = it title $ \ctx -> do
+    -- SETUP
+    let amnt = 100_000 :: Natural
+    wSrc <- fixtureSource ctx
+    (wDest, payment, mnemonics) <- do
+        (wDest, mnemonics) <- fixtureRandomWalletMws ctx
+        let addrs = randomAddresses @n mnemonics
+        pure (wDest, mkPayment @n (head addrs) amnt, mnemonics)
+
+    -- ACTION
+    r <- postByronTransaction @n ctx wSrc [payment] fixturePassphrase
+
+    -- ASSERTIONS
+    let (feeMin, feeMax) = ctx ^. #_feeEstimator $ PaymentDescription
+            { nInputs  = 1
+            , nOutputs = 1
+            , nChanges = 1
+            }
+    verify r
+        [ expectResponseCode HTTP.status202
+        , expectField #amount $ between
+              ( Quantity (feeMin + amnt)
+              , Quantity (feeMax + amnt)
+              )
+        , expectField #direction (`shouldBe` ApiT Outgoing)
+        , expectField #status (`shouldBe` ApiT Pending)
+        ]
+
+    eventually "source balance decreases" $ do
+        rSrc <- request @ApiByronWallet ctx
+            (Link.getWallet @'Byron wSrc) Default Empty
+        verify rSrc
+            [ expectField (#balance . #available) $ between
+                ( Quantity (faucetAmt - amnt - feeMax)
+                , Quantity (faucetAmt - amnt - feeMin)
+                )
+            ]
+
+    eventually "destination balance increases" $ do
+        rDest <- request @ApiByronWallet ctx
+            (Link.getWallet @'Byron wDest) Default Empty
+        verify rDest
+            [ expectField (#balance . #available)
+                (`shouldBe` Quantity (faucetAmt + amnt))
+            ]
+
+    rd1 <- request
+           @ApiByronWallet ctx (Link.deleteWallet @'Byron wDest) Default Empty
+    expectResponseCode @IO HTTP.status204 rd1
+
+    let (Right seed) = fromMnemonic @'[12] (mnemonicToText mnemonics)
+    let rootXPrv = T.decodeUtf8 $ hex $ getKey $
+            generateKeyFromSeed seed
+            (Passphrase $ BA.convert $ T.encodeUtf8 fixturePassphrase)
+    let passwHash =
+            "31347c387c317c2b6a6f747446495a6a566d586f43374c6c54425a576c\
+            \597a425834515177666475467578436b4d485569733d7c78324d646738\
+            \49554a3232507235676531393575445a76583646552b7757395a6a6a2f\
+            \51303054356c654751794279732f7662753367526d726c316c657a7150\
+            \43676d364e6758476d4d2f4b6438343265304b4945773d3d"
```

Comment:
Where does this hash come from :no_mouth: ?

Reply:
I added tests for "cardano-wallet" here : https://github.com/input-output-hk/cardano-wallet/pull/1476/files#diff-131f348ed80ae192131796acae9042efR273

Reply:
moved to DSL with proper comment : https://github.com/input-output-hk/cardano-wallet/pull/1476/commits/c8e89edd33a98ab4ef7152dfe5e744b19180706c